### PR TITLE
TECH-140: Create test configuration for "/authorization/oauth-details" endpoint

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
Create test configuration for "/authorization/oauth-details" endpoint.

Create test configuration for /authorization/oauth-details endpoint from https://github.com/GovStackWorkingGroup/bb-identity/pull/6.
Identify the critical APIs for Authentication, write the test plan and Cucumber tests, as well as supporting code to run those Cucumber tests.

TICKET: https://govstack-global.atlassian.net/browse/TECH-140

Also fixed typos in other test cases.

### CODE REVIEW

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]
As for the bodies we use in request I recommend using: Data Management feature from pactumJS : https://pactumjs.github.io/guides/data-management.html#data-management.
But it is to confirm with the others.

Please take a look at what I mentioned below, maybe I missed something.

From what I see you return an ```invalid_body``` error message, which I could not find in the SWAGGER.
image

Also I can see in successful return example that errors come as ```null```, you are expecting an empty array

File: test/features/support/ui_oauth_details.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        response: {
+          transactionId: "string",
+          authFactors: [
+            {
+              type: "PIN",
+            },
+          ],
+          essentialClaims: ["all claims"],
+        },
+        errors: [],
```
Comment (mgrochalskisoldevelo) [inline test/features/support/ui_oauth_details.js:95]: image


seems like ```null``` should be the default value, also to change in mocks
```[] -> null```

File: test/features/support/ui_oauth_details.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  () => "None of the required parameters was provided."
+);
+
+When("The user triggers an action without the required parameters", () =>
+  specUIOAuthDetails.post(baseUrl).withBody({})
+);
+
+Then(
+  "The result of an operation returns an error because none required parameters provided",
+  async () => await errorResultFunction("invalid_body")
```
Comment (mgrochalskisoldevelo) [inline test/features/support/ui_oauth_details.js:194]: image

There is no ```invalid_body``` in errors enum

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]

**Review by mgrochalskisoldevelo — APPROVED** [review]

**Comment by KarolinaKopacz** [comment]
I'm going to close this PR because endpoint no longer exist in the bb-identity API spec.

**Pull request closed without merging**